core/src/main/java/com/scalar/db/storage/cassandra/Cassandra.java

```diff
@@ -64,42 +63,13 @@ public Cassandra(DatabaseConfig config) {
             new CassandraAdmin(clusterManager, config),
             config.getTableMetadataCacheExpirationTimeSecs());
     operationChecker = new OperationChecker(metadataManager);
-
-    namespace = Optional.empty();
-    tableName = Optional.empty();
-  }
-
-  @Override
-  public void with(String namespace, String tableName) {
-    this.namespace = Optional.ofNullable(namespace);
-    this.tableName = Optional.ofNullable(tableName);
-  }
-
-  @Override
-  public void withNamespace(String namespace) {
-    this.namespace = Optional.ofNullable(namespace);
-  }
-
-  @Override
-  public Optional<String> getNamespace() {
-    return namespace;
-  }
-
-  @Override
-  public void withTable(String tableName) {
-    this.tableName = Optional.ofNullable(tableName);
-  }
-
-  @Override
-  public Optional<String> getTable() {
-    return tableName;
   }
 
   @Override
   @Nonnull
   public Optional<Result> get(Get get) throws ExecutionException {
     LOGGER.debug("executing get operation with " + get);
-    ScalarDbUtils.setTargetToIfNot(get, namespace, tableName);
+    get = copyAndSetTargetToIfNot(get);
```

Comment:
We copy the operation at the begging of the methods (`get()`, `scan()`, `put()`, `delete()`, `mutation()`).